### PR TITLE
Add system-characteristics 'has-assurance-level' constraints

### PR DESCRIPTION
**Pull request by Gabeblis**
# Committer Notes
{Please provide a description of what this PR accomplishes. Be sure to reference any issues addressed. If the PR is a work-in-progress submitted for early review, please submit the PR as a draft PR using the "Draft pull request" dropdown.}

### Description
Added system-characteristics 'has-assurance-level' constraints and tests for each of the constraints. These should be added as they are part of the effort write all of the constraints from the constraint tracker.  

### Changes
- Added constraints has-identity-assurance-level, has-authenticator-assurance-level, and has-federation-assurance-level to fedramp-external-constraints.xml.  
- Added pass and fail yaml tests for all of the above constraints.  
- Edited ssp-all-VALID.xml to ensure all constraints pass correctly.  

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
~~- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?~~
~~- [ ] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?~~

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 1]
```diff
@@ -59,9 +59,18 @@
...
             <message>A FedRAMP SSP information-type categorization requires a correct system attribute. FedRAMP only supports the system value 'https://doi.org/10.6028/NIST.SP.800-60v2r1'.</message>
             </expect>
-            <expect id="categorization-has-information-type-id" target="//system-characteristics" test="system-information/information-type/categorization/information-type-id" level="ERROR">
+            <expect id="categorization-has-information-type-id" target="./system-characteristics" test="system-information/information-type/categorization/information-type-id" level="ERROR">
             <message>A FedRAMP SSP information type categorization must have at least one information type identifier.</message>
             </expect>
+            <expect id="has-identity-assurance-level" target="./system-characteristics" test="prop[@name eq 'identity-assurance-level']" level="INFORMATIONAL">
+            <message>A FedRAMP SSP may have a Digital Identity Determination identity assurance level property.</message>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:66]: A recommendation for some adjustment on wording:

```suggestion
            <message>This FedRAMP SSP does define its NIST SP 800-63 identity assurance level (IAL).</message>
```

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 2]
```diff
@@ -59,9 +59,18 @@
...
+            </expect>
+            <expect id="has-authenticator-assurance-level" target="./system-characteristics" test="prop[@name eq 'authenticator-assurance-level']" level="INFORMATIONAL">
+            <message>A FedRAMP SSP may have a Digital Identity Determination authenticator assurance level property.</message>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:69]: A recommendation for some adjustment on wording:

```suggestion
            <message>This FedRAMP SSP does define its NIST SP 800-63 authenticator assurance level (AAL).</message>
```

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 3]
```diff
@@ -59,9 +59,18 @@
...
+            </expect>
+            <expect id="has-federation-assurance-level" target="./system-characteristics" test="prop[@name eq 'federation-assurance-level']" level="INFORMATIONAL">
+            <message>A FedRAMP SSP may have a Digital Identity Determination federation assurance level property.</message>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:72]: A recommendation for some adjustment on wording:

```suggestion
            <message>This FedRAMP SSP does define its NIST SP 800-63 federation assurance level (IAL).</message>
```

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]

**Comment by aj-stein-gsa** [comment]
@Gabeblis you cool with me committing those recs and merging in this PR after that or would you like to discuss?

**Comment by Gabeblis** [comment]
> @Gabeblis you cool with me committing those recs and merging in this PR after that or would you like to discuss?

Can we briefly discuss?

**Comment by aj-stein-gsa** [comment]
> > @Gabeblis you cool with me committing those recs and merging in this PR after that or would you like to discuss?
> 
> Can we briefly discuss?

Thanks for discussing, sir! I appreciate the conversation. I will ref #675 because we need to look at examples like this review to discuss wording of certain types of messages, and we need consistent info before (not after) constraint writing about that.

**Review by aj-stein-gsa — APPROVED** [review]

**Pull request merged**